user/api.go

```diff
@@ -635,7 +648,7 @@ func (a *Api) DeleteUser(res http.ResponseWriter, req *http.Request, vars map[st
 
 	pw := getGivenDetail(req)["password"]
 
-	if id != "" && pw != "" {
+	if id != "" && (td.IsServer || pw != "") {
```

Comment:
Looks like we're missing some tests on Server token + missing password.

Reply:
Test done